python/k4FWCore/utils.py

```diff
@@ -77,14 +80,39 @@ def load_file(opt_file: Union[TextIOWrapper, str, os.PathLike]) -> None:
         Exception: Any exception raised by the executed code will be propagated.
 
     """
+    # Cannot simply deepcopy globals. Hence, populate the necessary stuff
+    namespace = {
+        "__file__": __file__,
+        "__builtins__": __builtins__,
+        "__loader__": __loader__,
+    }
+
     if isinstance(opt_file, (str, os.PathLike)):
         with open(opt_file, "r") as file:
             code = file.read()
+            filename = file.name
+
+        module_name = Path(opt_file).stem
+        loader = SourceFileLoader(module_name, str(opt_file))
+
+        namespace.update(
+            {
+                "__file__": os.path.realpath(opt_file),
+                "__spec__": importlib.util.spec_from_loader(loader.name, loader),
+            }
+        )
     else:
+        warnings.warn(
+            "load_file will remove support for handling TextIOWrapper. Please switch to pasing os.PathLike",
+            FutureWarning,
+        )
```

Comment:
I am not sure this is used anywhere. It might be possible to remove this entirely and clean the implementation up a bit more.